agent/config/builder.go

```diff
@@ -1038,6 +1038,10 @@ func (b *Builder) Validate(rt RuntimeConfig) error {
 	if rt.NodeName == "" {
 		return fmt.Errorf("node_name cannot be empty")
 	}
+	if !validDNSre.MatchString(rt.NodeName) {
```

Comment:
Well, actually, having node names with dots works now (DNS and not http) and is thus not limited to 63 characters.